hazelcast/src/main/java/com/hazelcast/internal/management/TimedMemberStateFactory.java

```diff
@@ -244,195 +222,153 @@ protected void createNodeState(MemberStateImpl memberState) {
-        for (Map.Entry<String, LocalTopicStats> entry : topics.entrySet()) {
-            String name = entry.getKey();
+    private void handleReliableTopic(MemberStateImpl memberState, Config config,
+                                     Map<String, LocalTopicStats> topics) {
+        Set<String> topicsWithStats = createHashSet(topics.size());
+        for (String name : topics.keySet()) {
             if (config.findReliableTopicConfig(name).isStatisticsEnabled()) {
-                LocalTopicStats stats = entry.getValue();
-                memberState.putLocalReliableTopicStats(name, stats);
-                ++count;
+                topicsWithStats.add(name);
             }
         }
-        return count;
+        memberState.setReliableTopicsWithStats(topicsWithStats);
     }
 
-    private int handleTopic(MemberStateImpl memberState, int count, Config config, Map<String, LocalTopicStats> topics) {
-        for (Map.Entry<String, LocalTopicStats> entry : topics.entrySet()) {
-            String name = entry.getKey();
+    private void handleTopic(MemberStateImpl memberState, Config config, Map<String, LocalTopicStats> topics) {
+        Set<String> topicsWithStats = createHashSet(topics.size());
+        for (String name : topics.keySet()) {
             if (config.findTopicConfig(name).isStatisticsEnabled()) {
-                LocalTopicStats stats = entry.getValue();
-                memberState.putLocalTopicStats(name, stats);
-                ++count;
+                topicsWithStats.add(name);
             }
         }
-        return count;
+        memberState.setTopicsWithStats(topicsWithStats);
     }
 
-    private int handleQueue(MemberStateImpl memberState, int count, Config config, Map<String, LocalQueueStats> queues) {
-        for (Map.Entry<String, LocalQueueStats> entry : queues.entrySet()) {
-            String name = entry.getKey();
+    private void handleQueue(MemberStateImpl memberState, Config config, Map<String, LocalQueueStats> queues) {
+        Set<String> mapsWithStats = createHashSet(queues.size());
+        for (String name : queues.keySet()) {
             if (config.findQueueConfig(name).isStatisticsEnabled()) {
-                LocalQueueStats stats = entry.getValue();
-                memberState.putLocalQueueStats(name, stats);
-                ++count;
+                mapsWithStats.add(name);
             }
         }
-        return count;
+        memberState.setQueuesWithStats(mapsWithStats);
     }
 
-    private int handleMap(MemberStateImpl memberState, int count, Config config, Map<String, LocalMapStats> maps) {
-        for (Map.Entry<String, LocalMapStats> entry : maps.entrySet()) {
-            String name = entry.getKey();
+    private void handleMap(MemberStateImpl memberState, Config config, Map<String, LocalMapStats> maps) {
+        Set<String> queuesWithStats = createHashSet(maps.size());
```

Comment:
Minor: queues -> maps

Reply:
Addressed in 3bae9fb45929a28d57d909a33aa7192c11018b94